dbt/include/databricks/macros/materializations/incremental/strategies.sql

```diff
@@ -51,24 +51,32 @@
 {% macro get_insert_into_sql(source_relation, target_relation) %}
 
     {%- set common_columns = [] -%}
-    {%- set source_columns = adapter.get_columns_in_relation(source_relation) | map(attribute="name") -%}
-    {%- for dest_col in adapter.get_columns_in_relation(target_relation) -%}
-      {% if dest_col.name in source_columns %}
-          {%- if common_columns.append(dest_col) -%}{%- endif -%}
+    {%- set source_columns = adapter.get_columns_in_relation(source_relation) | map(attribute="quoted") | list -%}
+    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) | map(attribute="quoted") | list -%}
+    {{ log("source_columns: " ~ (source_columns | join(', '))) }}
+    {{ log("dest_columns: " ~ (dest_columns | join(', '))) }}
+    {%- for dest_col in dest_columns -%}
+      {%- if dest_col in source_columns %}
+        {%- do common_columns.append(dest_col) -%}
+      {%- else -%}
+        {%- do common_columns.append('NULL') -%}
       {%- endif -%}
     {%- endfor -%}
-    {%- set dest_cols_csv = common_columns | map(attribute='quoted') | join(', ') -%}
+    {%- set dest_cols_csv = dest_columns | join(', ') -%}
+    {%- set source_cols_csv = common_columns | join(', ') -%}
     insert into table {{ target_relation }} ({{ dest_cols_csv }})
-    select {{dest_cols_csv}} from {{ source_relation }}
```

Comment:
was this just wrong before?

Reply:
I merged a community PR a couple of weeks ago that tried to fix an issue where, when the on_schema_change strategy was 'ignore', it would fail for column mismatch between the target and the source (in contradiction to the dbt docs).  The change introduced a break for 'append' and 'sync_all_columns' when the incremental strategy is set to 'append'.  So until now, we could not pass the functional tests for 'append' and 'ignore' at the same time, but we didn't see it because our integration tests were modified to a weaker standard than the dbt-core tests.